### PR TITLE
fork checks and testnet upgrade

### PR DESCRIPTION
**Pull request by 0xZenus**

### CODE REVIEW

File: packages/tests/test/reya_network/general/General.fork.t.sol [hunk 1]
```diff
@@ -55,27 +55,27 @@ contract GeneralForkTest is ReyaForkTest, GeneralForkCheck {
 
         assertLe(block.timestamp - ONE_MINUTE_IN_SECONDS, ethUsdNodeOutput.timestamp);
         assertLe(ethUsdNodeOutput.timestamp, block.timestamp);
-        assertApproxEqAbsDecimal(ethUsdNodeOutput.price, 3500e18, 1000e18, 18);
```
Comment (avniculae) [inline packages/tests/test/reya_network/general/General.fork.t.sol:58]: shall we update prices instead?

File: packages/tests/test/reya_cronos/general/General.fork.t.sol [hunk 2]
```diff
@@ -46,22 +46,22 @@ contract GeneralForkTest is ReyaForkTest, GeneralForkCheck {
         NodeOutput.Data memory usdcUsdNodeOutput = IOracleManagerProxy(sec.oracleManager).process(sec.usdcUsdNodeId);
 
         assertLe(ethUsdNodeOutput.timestamp, block.timestamp);
-        assertApproxEqAbsDecimal(ethUsdNodeOutput.price, 3500e18, 1000e18, 18);
```
Comment (avniculae) [inline packages/tests/test/reya_cronos/general/General.fork.t.sol:49]: shall we replace prices instead?

File: packages/tomls/src/omnibus/reya_cronos.toml [hunk 3]
```diff
@@ -13,7 +13,7 @@ include = [
     "../collateral_pools/collateral_pool_1/testnet.toml",
     "../orders_gateway/testnet.toml",
 ]
-version = "1.0.2"
+version = "1.0.3"
```
Comment (avniculae) [inline packages/tomls/src/omnibus/reya_cronos.toml:16]: can you please update this PR as well? https://github.com/Reya-Labs/reya-deployments/pull/40

File: packages/tomls/src/omnibus/reya_cronos.toml [hunk 4]
```diff
@@ -46,7 +46,7 @@ exchangePassRouterPackage = "reya-exchange-pass-nft:1.0.0@router"
 ranksProxyPackage = "reya-ranks:1.0.0@proxy"
 ranksRouterPackage = "reya-ranks:1.0.0@router"
 ordersGatewayProxyPackage = "reya-orders-gateway:1.0.1@proxy"
-ordersGatewayRouterPackage = "reya-orders-gateway:1.0.2@router"
+ordersGatewayRouterPackage = "reya-orders-gateway:1.0.3@router"
```
Comment (avniculae) [inline packages/tomls/src/omnibus/reya_cronos.toml:49]: update fuzzing.toml as well

File: packages/tests/test/reya_network/general/General.fork.t.sol (hunk 1 above)
Comment (0xZenus) [reply packages/tests/test/reya_network/general/General.fork.t.sol:58]: I don't think this is future proof. Anytime market gets crazy, we end up changing the prices.

File: packages/tests/test/reya_cronos/general/General.fork.t.sol (hunk 2 above)
Comment (0xZenus) [reply packages/tests/test/reya_cronos/general/General.fork.t.sol:49]: same as above

File: packages/tomls/src/omnibus/reya_cronos.toml (hunk 4 above)
Comment (0xZenus) [reply packages/tomls/src/omnibus/reya_cronos.toml:49]: bf63b29

File: packages/tomls/src/omnibus/reya_cronos.toml (hunk 3 above)
Comment (0xZenus) [reply packages/tomls/src/omnibus/reya_cronos.toml:16]: done

**Review by avniculae — DISMISSED** [review]

**Review by ioanaroceanu2000 — APPROVED** [review]

File: packages/tests/src/interfaces/IOrdersGatewayProxy.sol [hunk 5]
```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (arturbeg) [inline packages/tests/src/interfaces/IOrdersGatewayProxy.sol:1]: why are we duplicating this, means every time we need to change smth for the order gateway interface have to come back here and adjust ?

File: packages/tests/src/utils/CoreCommandHashing.sol [hunk 6]
```diff
@@ -0,0 +1,76 @@
+pragma solidity >=0.8.19 <0.9.0;
+
+import { Command } from "../../src/interfaces/IPeripheryProxy.sol";
+
+library CoreCommandHashing {
```
Comment (arturbeg) [inline packages/tests/src/utils/CoreCommandHashing.sol:5]: do we need to copy paste the entire lib to introduce mock digest calculation?

File: packages/tests/test/reya_common/BaseReyaForkTest.sol [hunk 7]
```diff
@@ -105,11 +103,8 @@ contract BaseReyaForkTest is StorageReyaForkTest {
             exchangeId: s.exchangeId
         });
 
-        s.digest = mockCoreCalculateDigest(
-            sec.core,
-            hashExecuteBySigExtended(
-                address(sec.periphery), accountId, commands, incrementedNonce, s.deadline, keccak256(abi.encode())
-            )
+        s.digest = CoreCommandHashing.mockCalculateDigest(
```
Comment (arturbeg) [inline packages/tests/test/reya_common/BaseReyaForkTest.sol:106]: btw, given these are fork checks is mock... an appropriate way to name a function ? (given we're working against prod and wanting to simulate realistic scenarios)

**Review by arturbeg — APPROVED** [review]

File: packages/tests/src/interfaces/IOrdersGatewayProxy.sol (hunk 5 above)
Comment (0xZenus) [reply packages/tests/src/interfaces/IOrdersGatewayProxy.sol:1]: This is almost automated, we don't do any copy paste from reya-network, but get ABI from the current deployment and use abi-to-sol to generate these interfaces. We should be able to generate these interfaces in a CLI command and AN and me are aware of this but we need some time to do it (especially that abi-to-sol has some limitations regarding events and custom datatypes).

File: packages/tests/src/utils/CoreCommandHashing.sol (hunk 6 above)
Comment (0xZenus) [reply packages/tests/src/utils/CoreCommandHashing.sol:5]: yes, we do (we do in unit tests as well)

File: packages/tests/test/reya_common/BaseReyaForkTest.sol (hunk 7 above)
Comment (0xZenus) [reply packages/tests/test/reya_common/BaseReyaForkTest.sol:106]: yes, unfortunately it's hard to properly check EIP712 signatures in Solidity (from integration point of view) and this is why we need to generate them this way ... anyway I think mock prefix is not really good naming here tho (it's not mock but we somehow introduced this in core UTs and kept dragging it from there)

**Review by avniculae — APPROVED** [review]

**Pull request merged**